src/stan/io/deserializer.hpp

```diff
@@ -944,6 +944,301 @@ class deserializer {
     }
     return ret;
   }
+
+  /**
+   * Unconstrain lower bounded variable
+   *
+   * @tparam S Type of lower bounded input
+   * @tparam L Type of lower bound
+   * @param y Lower bounded input
+   * @param lb Lower bound
+   * @return Unconstrained variable
+   */
+  template <typename S, typename L>
+  inline auto free_lb(const S& y, const L& lb) {
+    return stan::math::lb_free(y, lb);
+  }
+
+  /**
+   * Unconstrain upper bounded variable
+   *
+   * @tparam S Type of upper bounded input
+   * @tparam U Type of upper bound
+   * @param y Upper bounded input
+   * @param ub Upper bound
+   * @return Unconstrained variable
+   */
+  template <typename S, typename U>
+  inline auto free_ub(const S& y, const U& ub) {
+    return stan::math::ub_free(y, ub);
+  }
+
+  /**
+   * Unconstrain bounded variable
+   *
+   * @tparam S Type of bounded input
+   * @tparam L Type of lower bound
+   * @tparam U Type of upper bound
+   * @param y Upper bounded input
+   * @param lb Lower bound
+   * @param ub Upper bound
+   * @return Unconstrained variable
+   */
+  template <typename S, typename L, typename U>
+  inline auto free_lub(const S& y, const L& lb, const U& ub) {
```

Comment:
[optional] @rybern is it okay that the name pattern here is

```
free_{constrain_name}(object, {constrain_params});
```

Or would it be easier for you if we had the params come first like in the constrain functions?

Also @bbbales2 should we name these `read_free_{constrain_name}` and change the constrain functions to `read_constrain_{constrain_name}`? That would just help keep the API more consistent

Reply:
That shouldn't make it any easier or harder for me, up to you.

This looks awesome!